welcome.cpp

```diff
@@ -7,4 +7,59 @@
  * 
  * @copyright Copyright (c) 2023
  * 
- */
\ No newline at end of file
+ */
+
+#include "role.h"
+#include "welcome.h"
+#include "rule.h"
+#include <ncurses.h>
+
+extern int MAX_X;
+extern int MAX_Y;
+extern void drawRuleScene();
+
+void drawWelcomeScene() {
+
+  initscr();
+  wresize(stdscr, MAX_Y, MAX_X);
+  box(stdscr, '*', '*');
+
+  int y = (MAX_Y - 7) / 2;  // 7 是 *tank 的行數
+  int x = (MAX_X - 23) / 2; // 23 是 *tank 中最長的一行的長度
+
+  Role tank(tankBody, 7, 23);
+  tank.draw(y, x);
+
+  refresh();
+  
+  while (true) {
+    int ch = getch();
+
+    if (ch == 'q') 
+      break;
+    else if (ch == 'n') 
+      drawRuleScene();
+      break;
```

Comment:
It's important to ensure the loop is properly terminated, otherwise the program may not end as intended.